test/unit/refinement-crumbs.ts

```diff
@@ -50,36 +81,64 @@ suite('RefinementCrumbs', ({ expect, spy, stub, itShouldProvideAlias }) => {
       expect(refinementCrumbs.previousField).to.eq(field);
     });
 
-    it('should remove old listener', () => {
+    it('should remove SELECTED_REFINEMENTS_UPDATED listener if storeSection is search', () => {
       const off = spy();
       const previousField = (refinementCrumbs.previousField = refinementCrumbs.previousField = 'brand');
-      refinementCrumbs.props = { field: 'material' };
+      refinementCrumbs.props.storeSection = StoreSections.SEARCH;
       refinementCrumbs.flux = { off, on: () => null } as any;
       refinementCrumbs.selectRefinements = () => null;
 
+      refinementCrumbs.init();
       refinementCrumbs.updateState();
 
       expect(off).to.be.calledWith(`${Events.SELECTED_REFINEMENTS_UPDATED}:${previousField}`);
     });
 
-    it('should listen for SELECTED_REFINEMENTS_UPDATED', () => {
-      const field = 'material';
+    it('should remove PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED listener if storeSection is pastpurchases', () => {
+      const off = spy();
+      const previousField = (refinementCrumbs.previousField = refinementCrumbs.previousField = 'brand');
+      refinementCrumbs.props.storeSection = StoreSections.PAST_PURCHASES;
+      refinementCrumbs.flux = { off, on: () => null } as any;
+      refinementCrumbs.selectRefinements = () => null;
+
+      refinementCrumbs.init();
+      refinementCrumbs.updateState();
+
+      expect(off).to.be.calledWith(`${Events.PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED}:${previousField}`);
+    });
+
+    it('should listen for SELECTED_REFINEMENTS_UPDATED if storeSection is search', () => {
       const on = spy();
-      refinementCrumbs.props = { field };
+      refinementCrumbs.props.storeSection = StoreSections.SEARCH;
       refinementCrumbs.previousField = 'brand';
       refinementCrumbs.flux = { on, off: () => null } as any;
       refinementCrumbs.selectRefinements = () => null;
 
+      refinementCrumbs.init();
       refinementCrumbs.updateState();
 
       expect(on).to.be.calledWith(`${Events.SELECTED_REFINEMENTS_UPDATED}:${field}`);
     });
+
+    it('should listen for PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED if storeSection is pastpurchases', () => {
+      const on = spy();
+      refinementCrumbs.props.storeSection = StoreSections.PAST_PURCHASES;
+      refinementCrumbs.previousField = 'brand';
+      refinementCrumbs.flux = { on, off: () => null } as any;
+      refinementCrumbs.selectRefinements = () => null;
+
+      refinementCrumbs.init();
+      refinementCrumbs.updateState();
+
+      expect(on).to.be.calledWith(`${Events.PAST_PURCHASE_SELECTED_REFINEMENTS_UPDATED}:${field}`);
+    });
   });
 
   describe('updateRefinements()', () => {
     it('should update state', () => {
       const newState: any = { a: 'b' };
       const set = (refinementCrumbs.set = spy());
+      refinementCrumbs.state = <any>{ navigationSelector: spy() }
```

Comment:
can just be `navigationSelector: () => null` don't need to use a spy because we're not checking if it was called